### PR TITLE
[fix] 281, make grafana links more prominent

### LINKED ISSUE #281: Make grafana links more prominent
Currently we are printing the links to grafana for each node in the spawn flow. To make more easy to check them we should also print at the end of the test/spawn with some legend like

```
Hey one of your test failed, to see the full logs of the nodes please go to 
alice: XY
bob: YX
...
```

**Update** Also, when the job/test fails we should grab all the node's logs and expose them as artifacts so external contributors can check them.

cc @bkchr

### PR DESCRIPTION
**Pull request by pepoviola**
fix #281

### CODE REVIEW

File: src/configGenerator.ts [hunk 1]
```diff
@@ -358,8 +362,10 @@ export async function generateBootnodeSpec(config: ComputedNetwork): Promise<Nod
     telemetryUrl: "",
     overrides: [],
     zombieRole: "bootnode",
-    imagePullPolicy: config.settings.image_pull_policy? config.settings.image_pull_policy : "Always",
-    ...ports
+    imagePullPolicy: config.settings.image_pull_policy
+      ? config.settings.image_pull_policy
+      : "Always",
```
Comment (wirednkod) [inline src/configGenerator.ts:367]:
```suggestion
    imagePullPolicy: config.settings.image_pull_policy || "Always",
```

File: src/configGenerator.ts [hunk 2]
```diff
@@ -453,8 +461,10 @@ async function getCollatorNodeFromConfig(
     overrides: [],
     zombieRole: cumulusBased ? "cumulus-collator" : "collator",
     parachainId: para_id,
-    imagePullPolicy: networkSpec.settings.image_pull_policy? networkSpec.settings.image_pull_policy : "Always",
-    ...ports
+    imagePullPolicy: networkSpec.settings.image_pull_policy
+      ? networkSpec.settings.image_pull_policy
+      : "Always",
```
Comment (wirednkod) [inline src/configGenerator.ts:466]:
```suggestion
    imagePullPolicy: networkSpec.settings.image_pull_policy || "Always",
```

File: src/configGenerator.ts [hunk 3]
```diff
@@ -536,20 +548,22 @@ async function getNodeFromConfig(
     addToBootnodes: node.add_to_bootnodes ? true : false,
     resources: node.resources || networkSpec.relaychain.defaultResources,
     zombieRole: "node",
-    imagePullPolicy: networkSpec.settings.image_pull_policy? networkSpec.settings.image_pull_policy : "Always",
-    ...ports
+    imagePullPolicy: networkSpec.settings.image_pull_policy
+      ? networkSpec.settings.image_pull_policy
+      : "Always",
```
Comment (wirednkod) [inline src/configGenerator.ts:553]:
```suggestion
    imagePullPolicy: networkSpec.settings.image_pull_policy || "Always",
```

File: src/test-runner/index.ts [hunk 4]
```diff
@@ -150,21 +150,83 @@ export async function run(
...
-          test.state !== "passed";
+          return test.state !== "passed";
         });
         if (fail) {
-          // keep the namespace up for 1 hour
           console.log(
-            `\n\t ${decorators.yellow(
-              "Some test fail, we will keep the namespace up for 30 more minutes",
+            `\n\n\t${decorators.red(
+              "Hey one of your test failed, to see the full logs of the nodes please go to:",
```
Comment (wirednkod) [inline src/test-runner/index.ts:162]: Is it always one? Or they can be more?

File: src/test-runner/index.ts [hunk 5]
```diff
@@ -150,21 +150,83 @@ export async function run(
...
+                    "Logs are also available in the artifacts' pipeline in gitlab",
+                  )}`,
+                );
+              } else {
+                console.log(`\n\t${decorators.magenta(logsPath)}`);
+              }
+              // keep pods running for 30 mins.
+              console.log(
+                `\n\t${decorators.yellow(
+                  "Some test fail, we will keep the namespace up for 30 more minutes",
```
Comment (wirednkod) [inline src/test-runner/index.ts:222]: Same here

File: src/utils/misc-utils.ts [hunk 6]
```diff
@@ -79,3 +80,17 @@ export function convertExponentials(data: string): string {
...
+export function getLokiUrl(
+  namespace: string,
+  podName: string,
+  from: number | string,
+  to?: number | string,
+): string {
+  const loki_url = LOKI_URL_FOR_NODE.replace(/{{namespace}}/, namespace)
+    .replace(/{{podName}}/, podName)
+    .replace(/{{from}}/, from.toString())
+    .replace(/{{to}}/, to ? to.toString() : "now");
```
Comment (wirednkod) [inline src/utils/misc-utils.ts:93]:
```suggestion
    .replace(/{{to}}/, to?.toString() || "now");
```

File: src/test-runner/index.ts (hunk 4 above)
Comment (pepoviola) [reply src/test-runner/index.ts:162]: Can be more, let me rephrase those :)

**Review by wirednkod — APPROVED** [review]
